apps/api/src/resources/swagger.yaml

```diff
@@ -148,8 +118,120 @@ paths:
+      responses:
+        '500':
+          description: 'Error creating application'
+          content:
+            application/json:
+              schema:
+                type: object
+        '400':
+          description: 'Error creating application - No User ID specified.'
+          content:
+            application/json:
+              schema:
+                type: object
+        '201':
+          description: 'Application created'
+          content:
+            application/json:
+              schema:
+                type: object
+                $ref: '#/components/schemas/Application'
+
+  /applications/edit:
+    post:
+      tags:
+        - Applications
+      summary: Edit application details
+      requestBody:
+        description: Changes made to an Application record
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ApplicationUpdates'
+      responses:
+        '500':
+          description: 'Error editing application'
+          content:
+            application/json:
+              schema:
+                type: object
+        '400':
+          description: 'Error editing application - Application not found'
+          content:
+            application/json:
+              schema:
+                type: object
+        '200':
+          description: 'Application updated'
+          content:
+            application/json:
+              schema:
+                type: object
+
 components:
   schemas:
+    Application:
+      type: object
+      properties:
+        id:
+          type: number
+          description: The unique ID of the application in the database.
+        user_id:
+          type: string
+          description: The user ID associated with the application.
+        state:
+          type: string
+          enum:
+            [DRAFT, INSTITUTIONAL_REP_REVIEW, DAC_REVIEW, DAC_REVISIONS_REQUESTED, REJECTED, APPROVED, CLOSED, REVOKED]
+          description: The current state that the application is in.
+        created_at:
+          type: string
+          format: date-time
+          description: When the application was created
+        approved_at:
+          type: string
+          format: date-time
+          nullable: true
+          description: When the application was approved. Can be NULL.
+        updated_at:
+          type: string
+          format: date-time
+          nullable: true
+          description: When the application was updated. Can be NULL.
+        expires_at:
+          type: string
+          format: date-time
+          nullable: true
+          description: When the application expires. Can be NULL.
+        contents:
+          type: number
+          description: A ID referring to the entry with the application contents.
+
+    ApplicationCreation:
+      type: object
+      required:
+        - user_id
+      properties:
+        user_id:
```

Comment:
There's a mismatch here with application router which is expecting `userId` in request.body

Reply:
shoot, will fix that! However, i wonder if we should standardize a convention for naming this? Internally we use `user_id`, but API / outward facing we use `userId`, should we just switch over to `user_id`?

Reply:
So the conflict is between JSX conventions (camelCase) vs Postgres conventions (snake case)
Which is why the UI request is typically formatted camelCase `userId` vs API snake case `user_id`

Now, in our argo clinical service, there's a fair bit of aliasing and equality management going on, you can see this file as an example:
https://github.com/icgc-argo/argo-clinical/blob/develop/src/common-model/entities.ts

We will have to standardize this, and we should add a note to a planned ticket on the roadmap.
It may be more relevant when we get to the DTO / Zod validation / error handling set up.
We could get a start on it at this point, but for the scope of this PR just changes to the names should be enough.